pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -517,6 +518,25 @@ func (configgen *ConfigGeneratorImpl) buildSidecarThriftListenerOptsForPortOrUDS
 	return thriftOpts
 }
 
+func (configgen *ConfigGeneratorImpl) buildSidecarInboudDubboListenerOptsForPortOrUDS(pluginParams *plugin.InputParams) *dubboListenerOpts {
+	// In case of unix domain sockets, the service port will be 0. So use the port name to distinguish the
+	// inbound listeners that a user specifies in Sidecar. Otherwise, all inbound clusters will be the same.
+	// We use the port name as the subset in the inbound cluster for differentiation. Its fine to use port
+	// names here because the inbound clusters are not referred to anywhere in the API, unlike the outbound
+	// clusters and these are static endpoint clusters used only for sidecar (proxy -> app)
+	clusterName := model.BuildSubsetKey(model.TrafficDirectionInbound, pluginParams.ServiceInstance.ServicePort.Name,
+		pluginParams.ServiceInstance.Service.Hostname, int(pluginParams.ServiceInstance.Endpoint.EndpointPort))
+
+	dubboOpts := &dubboListenerOpts{
+		statPrefix:        clusterName,
+		protocolType:      dubbo.ProtocolType_Dubbo,
+		serializationType: dubbo.SerializationType_Hessian2,
+		routeConfig:       configgen.buildSidecarDubboRouteConfig(clusterName, string(pluginParams.ServiceInstance.Service.Hostname)),
```

Comment:
Curious how the hostname equals interface that in real case?

Reply:
Interfaces are registered to Istio registry as services, which can be done through ServiceEntry. There're a few ways to accomplish that: One way is to create a dubbo interface CRD and use a controller to watch these CRD resources and sync them to Istio as ServiceEntry. Another way is to create an MCP server to connect the dubbo registry. We're using the first approach.